docs/broker/filtering.md

```diff
@@ -0,0 +1,612 @@
+
+```yaml
+spec:
+  filter:
+    expression: >
+      ce.type == "com.github.pull.create" &&
+      ce.source == "https://github.com/knative/eventing/pulls/123"
+```
+
+### Prefix match on source
+
+Cannot be specified with the SourceAndType filter or the attributes filter.
+
+CEL supports multiple options for prefix match. The simplest is `startsWith`:
+
+```yaml
+spec:
+  filter:
+    expression: ce.source.startsWith("https://github.com")
+```
+
+CEL also implements shell-style wildcards with `match`:
+
+```yaml
+spec:
+  filter:
+    expression: ce.source.match("https://github.com*")
+```
+
+For the most complex matching needs, CEL implements
+[RE2-style](https://github.com/google/re2/wiki/Syntax) regular expressions with
+`matches`:
+
+```yaml
+spec:
+  filter:
+    expression: ce.source.matches("https://github.com.*")
+```
+
+### Complex boolean expression
+
+Cannot be specified with the SourceAndType filter.
+
+_The `>` syntax is a standard yaml multiline string._
+
+```yaml
+spec:
+  filter:
+    expression: >
+      ce.type == "com.github.pull.create" ||
+      (ce.type == "com.github.issue.create" && ce.source.matches("proposals")
+```
+
+### Exact match on official and extension attribute
+
+Cannot be specified with the SourceAndType filter.
+
+_This example assumes the repository name is available as a CloudEvents
+extension `{"repository":proposals"}`._
+
+Specified with the attributes filter:
+
+```yaml
+spec:
+  filter:
+    attributes:
+      type: com.github.issue.create
+      repository: proposals
+```
+
+_The `>` syntax is a standard yaml multiline string._
+
+```yaml
+spec:
+  filter:
+    expression: >
+      ce.type == "com.github.issue.create" &&
+      ce.repository == "proposals"
+```
+
+### Exact match on extension with dotted name
+
+Cannot be specified with the SourceAndType filter.
+
+_This example assumes the repository name is available as a CloudEvents
+extension `{"github.repository":proposals"}`._
+
+Specified with the attributes filter:
+
+```yaml
+spec:
+  filter:
+    attributes:
+      github.repository: proposals
+```
+
+_The `>` syntax is a standard yaml multiline string._
+
+```yaml
+spec:
+  filter:
+    expression: >
+      ce["github.repository"] == "proposals"
+```
+
+### Exact match on nested extension
+
+Cannot be specified with the SourceAndType filter or the attributes filter.
+
+_This example assumes the repository name is available as a **nested**
+CloudEvents extension `{"github":{"repository":"proposals"}}`._
+
+_The `>` syntax is a standard yaml multiline string._
+
+```yaml
+spec:
+  filter:
+    expression: >
+      ce.github.repository == "proposals"
+```
+
+### Exact match on data string field
+
+Cannot be specified with the SourceAndType filter or the attributes filter.
+
+_This example assumes the event data is the parseable equivalent of
+`{"user":{"id":"abc123"}`._
+
+```yaml
+spec:
+  filter:
+    expression: >
+      user.id == "abc123"
+```
+
+### Exact match on data number field
```

Comment:
nit: Header says Exact match, but the example on L372 is a comparison. Given the L362 note, consider changing this to:
data.latency = 300.0

Though, are there rounding errors, or things like that to worry about?

Reply:
Good catch, I'll fix that.

In CEL numbers are 64 bit integers or doubles. I guess if you're doing arithmetic in the expression you might have to worry about rounding, but doesn't seem relevant if you're just comparing values.